### PR TITLE
Only check status if kited is not running

### PR DESCRIPTION
**Pull request by airblade**
See #283.

### CODE REVIEW

File: autoload/kite/status.vim [hunk 1]
```diff
@@ -33,6 +33,12 @@ function! kite#status#status(...)
     return
   endif
 
+  " If previously Kite wasn't logged in / installed / running, and now it is,
+  " trigger BufEnter to activate completion etc.
+  if index(['Kite: not logged in', 'Kite: not installed', 'Kite: not running'], getbufvar(buf, 'kite_status')) != -1
```
Comment (metalogical) [inline autoload/kite/status.vim:38]: nit: `Kite: not logged in` is no longer a vaild status

Comment (airblade) [reply autoload/kite/status.vim:38]: Do we no longer even check whether the user is logged in?

**Comment by airblade** [comment]
@airblade That's the behaviour this change gives us.  Here's the flow:

- When entering a buffer, initialise.
- When initialising, if Kite isn't running, do nothing except start the status polling.
- In the status polling, if we detect Kite is now running when it wasn't before, initialise the buffer.

**Comment by airblade** [comment]
However I have just realised that the status polling is only active if the user shows Kite's status in their status line.  We need the plugin to activate itself for a buffer regardless of whether the user shows Kite's status.

**Comment by airblade** [comment]
The updated flow is:

- When entering a buffer, initialise.
- If Kite isn't running, start watching for it.  Also start the status polling.
- In the watcher, when Kite becomes available re-initialise the buffer.

File: autoload/kite/status.vim (hunk 1 above)
Comment (jansorg) [reply autoload/kite/status.vim]: as discussed: Remove the login check for the status line

Comment (airblade) [reply autoload/kite/status.vim]: Done in 9a35159.

**Comment by airblade** [comment]
@jansorg The tests are failing because kited is not available in the test environment, and the plugin deactivates itself until kited becomes available.

**Comment by airblade** [comment]
@jansorg The tests pass now.

**Pull request merged**